burnman/minerals/JH_2015.py

```diff
@@ -198,30 +223,80 @@ class fm(CombinedMineral):
-                                 mineral_list = [di()],
+                                 mineral_list = [HP_2011_ds62.di()],
                                  molar_amounts = [1.],
                                  free_energy_adjustment=[-0.1e3, -0.211, 0.005e-5]) # note sign of *entropy* change.
 class cren(CombinedMineral):
     def __init__(self):
         CombinedMineral.__init__(self,
                                  name = 'chromium enstatite',
-                                 mineral_list = [mgts(), kos(), jd()],
+                                 mineral_list = [HP_2011_ds62.mgts(),
+                                                 HP_2011_ds62.kos(),
+                                                 HP_2011_ds62.jd()],
                                  molar_amounts = [1., 1., -1.],
                                  free_energy_adjustment=[3.e3, 0., 0.])
 
 class mess(CombinedMineral):
     def __init__(self):
         CombinedMineral.__init__(self,
                                  name = 'ferrienstatite',
-                                 mineral_list = [mgts(), acm(), jd()],
+                                 mineral_list = [HP_2011_ds62.mgts(),
+                                                 HP_2011_ds62.acm(),
+                                                 HP_2011_ds62.jd()],
                                  molar_amounts = [1., 1., -1.],
                                  free_energy_adjustment=[-15.e3, 0., 0.15e-5])
 
-        
+
+"""
+VARIANCE-COVARIANCE MATRIX
+Derived from HP_2011_ds62, modified to include all the CombinedMinerals
+
+cov is a dictionary containing:
+     - names: a list of endmember names
+     - covariance_matrix: a 2D variance-covariance array for the endmember enthalpies of formation
+"""
+# Create covariance matrix including all of the CombinedMinerals
+
+# List of combined endmembers in this dataset
+combined_endmembers = [cfs(),
+                       crdi(),
+                       cess(),
+                       cen(),
+                       cfm(),
+                       fm(),
+                       odi(),
+                       cren(),
+                       mess()]
+
+
+cov = {'endmember_names': copy(HP_2011_ds62.cov['endmember_names'])}
+
+A = np.identity(len(HP_2011_ds62.cov['endmember_names']))
+
+# Update names
+for c in combined_endmembers:
+    cov['endmember_names'].append(c.name)
+
+
+# Update covariance matrix
+for i, indices in enumerate([[HP_2011_ds62.cov['endmember_names'].index(name)
+                              for name in [mbr[0].params['name']
+                                           for mbr in c.mixture.endmembers]]
+                             for c in combined_endmembers]):
+    B = np.zeros(len(HP_2011_ds62.cov['endmember_names']))
+    B[indices] = combined_endmembers[i].mixture.molar_fractions
+    A = np.vstack((A, B))
+
+cov['covariance_matrix'] = A.dot(HP_2011_ds62.cov['covariance_matrix']).dot(A.T)
+
+del A
```

Comment:
wouldn't it be neater to create a function and call it to fill ``cov`` instead of executing code and deleting variables?